MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -251,15 +252,42 @@ public Task SaveImage(BaseItem item, Stream source, string mimeType, ImageType t
         }
 
         /// <inheritdoc/>
-        public Task SaveImage(BaseItem item, string source, string mimeType, ImageType type, int? imageIndex, bool? saveLocallyWithMedia, CancellationToken cancellationToken)
+        public async Task SaveImage(BaseItem item, string source, string mimeType, ImageType type, int? imageIndex, bool? saveLocallyWithMedia, CancellationToken cancellationToken)
         {
             if (string.IsNullOrWhiteSpace(source))
             {
                 throw new ArgumentNullException(nameof(source));
             }
 
-            var fileStream = AsyncFile.OpenRead(source);
-            return new ImageSaver(_configurationManager, _libraryMonitor, _fileSystem, _logger).SaveImage(item, fileStream, mimeType, type, imageIndex, saveLocallyWithMedia, cancellationToken);
+            ExceptionDispatchInfo? saveException = null;
+
+            try
+            {
+                var fileStream = AsyncFile.OpenRead(source);
+                await new ImageSaver(_configurationManager, _libraryMonitor, _fileSystem, _logger).SaveImage(item, fileStream, mimeType, type, imageIndex, saveLocallyWithMedia, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                saveException = ExceptionDispatchInfo.Capture(ex);
```

Comment:
Why not just let it throw? The finally block will get executed regardless. If `File.Delete` throws a non-IOException, the exception by ImageSaver will get swallowed

Reply:
I was not quite confident with the behavior of just throw without catching for such nested try catch. So we can still see if it is from the `ImageSaver` or from the `File.Delete` right? Because if the `ImageSaver` failed for some reason the `File.Delete` would also have a high chance to fail as well, we need to tell which one caused the failure.